.dotnet/src/Custom/Chat/ChatMessageContentPart.cs

```diff
@@ -4,170 +4,144 @@
-    /// </summary>
+    /// <summary> The image bytes. </summary>
+    /// <remarks> Present when <see cref="Kind"/> is <see cref="ChatMessageContentPartKind.Image"/>. </remarks>
     public BinaryData ImageBytes => _imageUrl?.ImageBytes;
 
-    /// <summary>
-    /// The image URI content.
-    /// </summary>
+    /// <summary> The MIME type of the image, e.g., <c>image/png</c>. </summary>
+    /// <remarks> Present when <see cref="Kind"/> is <see cref="ChatMessageContentPartKind.Image"/>. </remarks>
     public string ImageBytesMediaType => _imageUrl?.ImageBytesMediaType;
 
     /// <summary>
-    /// The image URI detail.
-    /// </summary>
-    public ImageChatMessageContentPartDetail? ImageDetail => _imageUrl?.Detail;
-
-    /// <summary>
-    /// Creates a new instance of <see cref="ChatMessageContentPart"/> that encapsulates text content.
+    ///     The level of detail with which the model should process the image and generate its textual understanding of
+    ///     it. Learn more in the <see href="https://platform.openai.com/docs/guides/vision/low-or-high-fidelity-image-understanding">vision guide</see>.
     /// </summary>
-    /// <param name="text"> The content for the new instance. </param>
-    /// <returns> A new instance of <see cref="ChatMessageContentPart"/>. </returns>
-    public static ChatMessageContentPart CreateTextMessageContentPart(string text)
-    {
-        Argument.AssertNotNull(text, nameof(text));
+    /// <remarks> Present when <see cref="Kind"/> is <see cref="ChatMessageContentPartKind.Image"/>. </remarks>
+    public ChatImageDetailLevel? ImageDetailLevel => _imageUrl?.Detail;
 
-        return new(text);
-    }
+    /// <summary> The refusal message generated by the model. </summary>
+    /// <remarks> Present when <see cref="Kind"/> is <see cref="ChatMessageContentPartKind.Refusal"/>. </remarks>
+    public string Refusal => _refusal;
 
-    /// <summary>
-    /// Creates a new instance of <see cref="ChatMessageContentPart"/> that encapsulates an assistant refusal message.
-    /// </summary>
-    /// <param name="refusal"> The refusal message from the assistant. </param>
-    /// <returns> A new instance of <see cref="ChatMessageContentPart"/>. </returns>
-    public static ChatMessageContentPart CreateRefusalMessageContentPart(string refusal)
+    /// <summary> Creates a new <see cref="ChatMessageContentPart"/> that encapsulates text. </summary>
+    /// <param name="text"> The text. </param>
+    public static ChatMessageContentPart CreateTextPart(string text)
     {
-        Argument.AssertNotNull(refusal, nameof(refusal));
+        Argument.AssertNotNull(text, nameof(text));
 
-        return new ChatMessageContentPart(
-            ChatMessageContentPartKind.Refusal.ToString(),
-            text: null,
-            refusal: refusal,
-            imageUrl: null,
-            serializedAdditionalRawData: null);
+        return new()
+        {
+            _kind = ChatMessageContentPartKind.Text,
```

Comment:
why do we do this instead of making the fields readonly and using a non-public constructor?